opm/grid/utility/ElementChunks.hpp

```diff
@@ -71,18 +71,21 @@ namespace Opm
 ///         // Do something else with elem
 ///     }
 /// }
-template <class GridView>
+template <class GridView, class PartitionSet>
 class ElementChunks
 {
 private:
-    using Iter = typename GridView::template Codim<0>::Iterator;
+    using Iter = decltype(std::begin(elements(std::declval<GridView>(), PartitionSet())));
```

Comment:
You need to be a little more careful here.  `Declval<>()` will (pretend to) create the **literal** type of its template argument&ndash;in this case `GridView`&ndash;whereas the `Iter` type in the `Chunk`s is the result of traversing a `const GridView&`.  The two iterator types may or may not be directly interchangeable, so I'd recommend writing this as
```C++
std::declval<const GridView&>()
```
instead.  As an aside, you should probably include `<iterator>` to bring in a declaration of `std::begin()` even though we **technically** get that from `<vector>` as well.

Reply:
Done.